src/components/VirtualizedAutocomplete/VirtualizedAutocomplete.stories.tsx

```diff
@@ -25,69 +24,28 @@ export const Default: VirtualizedStory = {
 	},
 };
 
-// TODO: FIGURE OUT WHY THIS IS NOT WORKING
-const VirtualizedPaginatedTemplate = <
-	Value,
-	Multiple extends boolean | undefined = false,
-	DisableClearable extends boolean | undefined = false,
-	FreeSolo extends boolean | undefined = false,
-	ChipComponent extends ElementType = ChipTypeMap['defaultComponent'],
->(
-	args: Omit<
-		VirtualizedAutocompleteWithPaginationProps<
+type VirtualizedPaginatedStory = StoryObj<
+	Meta<
+		<
 			Value,
-			Multiple,
-			DisableClearable,
-			FreeSolo,
-			ChipComponent
-		>,
-		'loadNext'
-	>,
-) => {
-	const [prevQuery, setPrevQuery] = useState('');
-	const [paginatedArray, setPaginatedArray] = useState<Value[]>([]);
-	const [page, setPage] = useState(-1);
-
-	return (
-		<VirtualizedAutocomplete<
-			Value,
-			Multiple,
-			DisableClearable,
-			FreeSolo,
-			ChipComponent
-		>
-			{...args}
-			loadNext={async (pageToLoad, query) => {
-				if (pageToLoad > page) {
-					setPage(pageToLoad);
-					setPaginatedArray(
-						paginatedArray.concat(
-							Array.from({ length: 50 }, () =>
-								Math.random().toString(),
-							) as Value[],
-						),
-					);
-				}
-				if (query !== prevQuery) {
-					setPrevQuery(query ?? '');
-					const filteredArray = paginatedArray.filter((option) =>
-						JSON.stringify(option).includes(query ?? ''),
-					);
-					return { data: filteredArray, totalItems: filteredArray.length + 1 };
-				}
-				return { data: paginatedArray, totalItems: paginatedArray.length + 1 };
-			}}
-		/>
-	);
-};
```

Comment:
wouldn't it be better to keep the TODO and analyse why this is not working when we have time ?